### ISSUE TITLE
Feature request: Support reading secrets from terminal in kubectl

### ISSUE THREAD

**Issue by tamalsaha** [kubernetes/kubernetes#54200]
`kubectl create secret` command supports `--from-literal` and `--from-file` options. But neither options are convenient when creating secrets manually from terminal.

- Using -- from-literal will keep the secret in shell history.
- Using --from-file requires creating the file first and then deleting it, etc.

I would like to see a flag like `--from-keys=k1,k2,k3`. Given this flag is present, kubectl will ask user for input for each key and create a secret using that info. It can use something like https://github.com/howeyc/gopass to read the value without echo.

/kind feature
/sig cli

**Comment by php-coder** [comment]
Because in Unix terminal is represented as a file `/dev/stdin`, you can use the following command:
```console
$ kubectl create secret generic foo --from-file=test=/dev/stdin
tada <-- this was manually typed in a terminal
secret "foo" created
$ kubectl get secret foo -o yaml | grep ^data: -A1
data:
  test: dGFkYQo=
$ echo -n dGFkYQo= | base64 -d
tada
```

**Comment by tamalsaha** [comment]
Thanks @php-coder ! I guess this works in linux / mc. Windows users will be out of luck.

**Comment by tamalsaha** [comment]
@php-coder , I just tried this in Ubuntu 16.04. How to I end the read process? I tried pressing `Enter` but the read did not close.

**Comment by php-coder** [comment]
> I tried pressing Enter but the read did not close.

Yes, I forgot to mention that you should use `Ctrl+D` to quit :)

**Comment by tamalsaha** [comment]
:) It worked after ctrl + D. I even tried with multiple keys.

I think one issue still remains in that I don't know in what order the keys are read by kubectl. A common scenario is creating AWS credentials. If kubectl can promt back the secret, that will make things bit nicer.

**Comment by sureshamk** [comment]
I use this handy method 

`kubectl get secrets <secretname>  -o json  | jq '.data | map_values(@base64d)'`
